catalog/transformer/catalog-transformer-zip/src/main/java/org/codice/ddf/catalog/transformer/zip/ZipCompression.java

```diff
@@ -240,27 +347,36 @@ private void writeResourceToZip(
 
   private BinaryContent getBinaryContentFromZip(String filePath)
       throws CatalogTransformerException {
-    BinaryContent binaryContent;
-    InputStream fileInputStream;
+    byte[] bytes;
     File zipFile = new File(filePath);
     try {
-      fileInputStream = new ZipInputStream(new FileInputStream(zipFile));
-      binaryContent = new BinaryContentImpl(fileInputStream);
-    } catch (FileNotFoundException e) {
-      throw new CatalogTransformerException("Unable to get ZIP file from ZipInputStream.", e);
+      bytes = Files.readAllBytes(zipFile.toPath().toAbsolutePath());
+    } catch (IOException e) {
+      throw new CatalogTransformerException(
+          "An error occurred while streaming the temporary zip file", e);
     }
 
-    jarSigner.signJar(
-        zipFile,
-        AccessController.doPrivileged(
-            (PrivilegedAction<String>) () -> System.getProperty(SystemBaseUrl.EXTERNAL_HOST)),
-        AccessController.doPrivileged(
-            (PrivilegedAction<String>) () -> System.getProperty("javax.net.ssl.keyStorePassword")),
-        AccessController.doPrivileged(
-            (PrivilegedAction<String>) () -> System.getProperty("javax.net.ssl.keyStore")),
-        AccessController.doPrivileged(
-            (PrivilegedAction<String>) () -> System.getProperty("javax.net.ssl.keyStorePassword")));
-    return binaryContent;
+    AccessController.doPrivileged(
```

Comment:
❓ Do we really need to sign the jar?  I realize this is existing but not entirely sure we want to take the performance hit for this and not sure it is needed.  Curious about @rzwiefel @bdeining thoughts here

Reply:
Also if the JDK isn't installed on the machine that this runs on it won't work because `jarSigner` is a part of the JDK and not the JRE

Reply:
This flow seems very odd to me.  I don't understand why the file that is written to the file system gets signed after the content is read in to be returned by the transformer.  That means the signature is not in the returned data.  I realize that this was pre-existing but I would argue that this needs to be reworked.  If we want a signed zip that should be handled in a different way.  I also think that passing in a file path to write it out to seems like an odd use of a transformer and kind of violates the contract of a transformer.  Really, it feels like a hack.  Does anyone actually use this "feature"?

Reply:
So I believe the signing is here, because this transformer is used when exporting data during an upgrade and we check for a signed zip file when importing data to a new system, so that we know it wasn't tampered with. I don't think it is necessary to sign zips that are just returning data that folks have queried. That being said, if you remove it, then you need to fix the upgrade case somehow to ensure that data is not tampered when performing an upgrade.

Reply:
I'm going to move the existing zip compression functionality into `DumpCommand.java` so it won't be necessary in this file anymore.